app/services/import_cait_indc.rb

```diff
@@ -0,0 +1,170 @@
+META_INDICATORS_FILEPATH = 'cait_indc/Backend-CAIT INDC Map - indicators.csv'.
+  freeze
+META_LEGEND_FILEPATH = 'cait_indc/Backend-CAIT INDC Map - legend.csv'.
+  freeze
+DATA_FILEPATH = 'cait_indc/Backend-CAIT INDC Map - data.csv'.
+  freeze
+
+class ImportCaitIndc
+  def call
+    cleanup
+
+    load_csvs
+
+    import_categories
+    import_indicator_types
+    import_charts
+    import_indicators
+
+    load_indicator_keys
+
+    import_labels
+    import_location_data
+    import_values
+  end
+
+  private
+
+  def load_csvs
+    @indicators = S3CSVReader.read(META_INDICATORS_FILEPATH).map(&:to_h)
+    @legend = S3CSVReader.read(META_LEGEND_FILEPATH).map(&:to_h)
+    @data = S3CSVReader.read(DATA_FILEPATH).map(&:to_h)
+  end
+
+  def load_indicator_keys
+    @indicator_keys = CaitIndc::Indicator.all.map do |i|
+      i.name.downcase.strip.tr(' ', '_').gsub(/\W/, '').to_sym
+    end
+  end
+
+  def cleanup
+    CaitIndc::LocationDatum.delete_all
+    CaitIndc::Value.delete_all
+    CaitIndc::Label.delete_all
+    CaitIndc::Indicator.delete_all
+    CaitIndc::Chart.delete_all
+    CaitIndc::Category.delete_all
+    CaitIndc::IndicatorType.delete_all
+  end
+
+  def chart(indicator)
+    chart_name = @legend.
+      detect { |l| l[:indicator_name] == indicator[:indicator] }&.
+      fetch(:chart_title, nil)
+
+    CaitIndc::Chart.find_by(name: chart_name) if chart_name
+  end
+
+  def indicator_attributes(indicator)
+    {
+      chart: chart(indicator),
+      indicator_type: CaitIndc::IndicatorType.
+        find_by(name: indicator[:indicator_type]),
+      category: CaitIndc::Category.find_by(name: indicator[:category]),
+      name: indicator[:indicator],
+      summary_list: indicator[:summary_list] == 'Yes',
+      on_map: indicator[:on_map] == 'Yes',
+      omit_from_detailed_view: indicator[:omit_from_detailed_view] == 'x',
+      show_in_dashboard: indicator[:show_in_dashboard] == 'x'
+    }
+  end
+
+  def label_attributes(label)
+    {
+      indicator: CaitIndc::Indicator.
+        find_by(name: label[:indicator_name]),
+      name: label[:legend_item_name],
+      color: label[:color]
+    }
+  end
+
+  def location_datum_attributes(datum)
+    {
+      location: Location.find_by(wri_standard_name: datum[:country]),
```

Comment:
this can hopefully become an iso code lookup once data is amended

Reply:
yep, I'm waiting on that